### ISSUE TITLE
slave can not reconnect to master

### ISSUE THREAD

**Issue by GoogleCodeExporter** [ashish21590/redis#371]
```
What steps will reproduce the problem?
one master and serveral slaves start on different sit.
Running for some time, the connction from slave to the master losts, 
and do not try connecting again.

What is the expected output? What do you see instead?


What version of the product are you using? On what operating system?
redis:2.0.0rc4 and 2.0.3
os:ubuntu9.1.0

Please provide any additional information below.
N/A

```

Original issue reported on code.google.com by `Mxnmw...@163.com` on 10 Nov 2010 at 2:33

**Comment by GoogleCodeExporter** [comment]
```
I had this problem too. Redis 2.2.2 would report the connection to the master 
server as up and running, but any commands would just hang. I think it was a 
generic firewall issue, because ssh connections would also just stop responding 
without disconnecting after a time of inactivity. 

The solution was to create a cron job that incremented a value in the master 
database that would keep the connection alive with the slaves. It would 
probably also be possible to use the "ping" command on each slave to connect to 
the master server.

I don't think it's possible to fix this in another way without implementing a 
new feature in Redis that would send keep alive requests in regular intervals.
```

Original comment by `norep2...@gmail.com` on 3 Apr 2011 at 9:54

**Comment by GoogleCodeExporter** [comment]
```
Sorry, my mistake. We were using version 2.0.4 :-)
```

Original comment by `norep2...@gmail.com` on 4 Apr 2011 at 7:36

**Comment by GoogleCodeExporter** [comment]
```
Not sure if this is the same problem that the original author had, but assuming 
it is: this has been fixed in Redis 2.2. The master sends an occasional ping to 
its slaves to ensure the connection does not time out. 

See: 
https://github.com/antirez/redis/commit/db0e263b428a33d458f4708eef9d9e9b974ce85a
```

Original comment by `pcnoordh...@gmail.com` on 4 Apr 2011 at 7:53
- Changed state: **Fixed**
